filter_finalreport.py

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env python3
+
+import argparse
+import glob
+
+
+def slice_file(input_file, chromosomes):
+    write_file = open("{}_sliced.txt".format(input_file[:-4]), "w")
+    input_lines = open(input_file, "r")
+    chrom_list = chromosomes.split(",")
+
+    """ Write all lines until Data section """
+    line = input_lines.readline()
+    while line.rstrip() != '[Data]':
+        write_file.write(line)
+        line = input_lines.readline()
+    write_file.write(line)  # Print [Data] line
+
+    """ Index header line for chromosome column, and write header"""
+    header = input_lines.readline()
+    chromosome_index = header.split("\t").index("Chr")
+    write_file.write(header)
+
+    """ Write all variant lines are on chromosomes of interest"""
+    for line in input_lines:
+        splitline = line.rstrip().split("\t")
+        if splitline[chromosome_index] in chrom_list:
+            write_file.write(line)
+
+    write_file.close()
```

Comment:
from python 2.5 and higher it is better practice to use `with open()` to avoid needing to close everything. especially using python 3. It will automatically close.

```
with open("hello.txt") as f:
    for line in f:
        print line
```
is the same as
```
f = open("hello.txt")
try:
    for line in f:
        print line
finally:
    f.close()
```

Reply:
Good point. Changed.